flow/flow.go

```diff
@@ -17,73 +17,73 @@ type Ckflow C.kflow
 const MAX_CUSTOM_STR_LEN = 384
 
 type Flow struct {
-	TimestampNano      int64
-	DstAs              uint32
-	DstGeo             uint32
-	DstMac             uint32 // IGNORED - use DstEthMac
-	HeaderLen          uint32
+	SrcBgpCommunity    string
+	SrcFlowTags        string
+	DstFlowTags        string
+	FlowTags           string
+	DstBgpAsPath       string
+	DstBgpCommunity    string
+	SrcBgpAsPath       string
+	Ipv6SrcAddr        []byte
+	Ipv6SrcRoutePrefix []byte
+	Customs            []Custom
+	Ipv6DstRoutePrefix []byte
+	Ipv6DstAddr        []byte
+	Ipv6SrcNextHop     []byte
+	Ipv6DstNextHop     []byte
```

Comment:
Looking at https://cs.opensource.google/go/x/tools/+/refs/tags/v0.9.3:go/analysis/passes/fieldalignment/fieldalignment.go;l=23-49

```
This analyzer find structs that can be rearranged to use less memory, and provides
a suggested edit with the most compact order.

Note that there are two different diagnostics reported. One checks struct size,
and the other reports "pointer bytes" used. Pointer bytes is how many bytes of the
object that the garbage collector has to potentially scan for pointers, for example:

	struct { uint32; string }

have 16 pointer bytes because the garbage collector has to scan up through the string's
inner pointer.

	struct { string; *uint32 }

has 24 pointer bytes because it has to scan further through the *uint32.

	struct { string; uint32 }

has 8 because it can stop immediately after the string pointer.

Be aware that the most compact order is not always the most efficient.
In rare cases it may cause two variables each updated by its own goroutine
to occupy the same CPU cache line, inducing a form of memory contention
known as "false sharing" that slows down both goroutines.
```

I guess it's the difference between pointer bytes and struct size. It would probably be good to try out the different options internally via some actual data and see how they compare before bringing it back up here unless it is well known that this current strategy works better. I assume based on the libraries used that it _should_ be good, but if we're doing this optimization we might as well be sure.